basic-api.yml

```diff
@@ -0,0 +1,142 @@
+openapi: 3.0.0
+info:
+  description: This is an open HTTP API specification for issuing Verifiable Credentials based on the W3C Verifiable Credentials Data Model (https://www.w3.org/TR/vc-data-model/). Implementations of this API may differ in terms of which representation formats or proof formats they do or do not support. Implementations MAY choose to not support certain parts of this API (e.g. optional parts include changing and retrieving the status of a credential, or refreshing a credential).
+  version: "0.0.2"
+  title: Verifiable Credential Issuer HTTP API
+  contact:
+    email: msporny@digitalbazaar.com
+  license:
+    name: All documents in this Repository are licensed by contributors under the [W3C Software and Document License](http://www.w3.org/Consortium/Legal/copyright-software).
+paths:
+  /credentials:
+    post:
+      summary: Issues a verifiable credential and returns it in the response body.
+      operationId: issueCredential
+      description: Issues a credential and returns it in the response body. Support of this part of the API is REQUIRED for implementations.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/IssueRequest'
+        description: Parameters for issuing the credential.
+      responses:
+        '201':
+          description: Credential was successfully issued.
+          content:
+            application/ld+json:
+              schema:
+                type: object
+                description: The issued Verifiable Credential, which is wrapped in a Verifiable Presentation.
+              example: {
+  "@context": [
+    "https://www.w3.org/2018/credentials/v1"
+  ],
+  "id": "urn:uuid:3978344f-8596-4c3a-a978-8fcaba3903c5",
+  "type": ["VerifiablePresentation"],
+  "verifiableCredential": [{
+    "@context": [
+      "https://www.w3.org/2018/credentials/v1",
+      "https://w3id.org/citizenship/v1"
+    ],
+    "id": "https://issuer.oidp.uscis.gov/credentials/83627465",
+    "type": ["VerifiableCredential", "PermanentResidentCard"],
+    "issuer": "did:v1:test:nym:z6Mko5ZVKPDsuTQN...46ecbs1u75hk2RJgAP7XU",
+    "issuanceDate": "2019-12-03T12:19:52Z",
+    "expirationDate": "2029-12-03T12:19:52Z",
+    "credentialSubject": {
+      "id": "did:example:b34ca6cd37bbf23",
+      "type": ["PermanentResident", "Person"],
+      "givenName": "JOHN",
+      "familyName": "SMITH",
+      "gender": "Male",
```

Comment:
this should be "sex" unless you want a 18 value list here :)

Reply:
> this should be "sex" unless you want a 18 value list here :)

Haha, well, the Citizenship vocabulary uses gender because schema.org uses gender:

https://digitalbazaar.github.io/citizenship-vocab/#gender

... and schema.org/gender allows for non-binary values. :)

... which just begs the question on whether or not federal governments allow non-binary values. I expect some will in time... we'll ask our customers what their position is on this. I expect applications can constrain the range of values if they want to.